### PR TITLE
[Silabs] Refactor/Cleanup Si917 examples build files

### PR DESCRIPTION
**Pull request by sharad-patil24**
Silabs Sample apps GN build is convoluted and duplicates a lot of args and conditions in each example's BUILD.gn.
This PR aims to reduce this duplication, regroup board or sdk arguments, and common example platform sources building to a common location.

Platforms affected by the refactoring:
SiWx917 ( SOC )

------------------------------
adds fix to the Bluez disconnection as well in separate commit.

### CODE REVIEW

File: src/platform/silabs/Logging.cpp [hunk 1]
```diff
@@ -431,7 +431,7 @@ extern "C" __attribute__((naked)) void HardFault_Handler(void)
                    "bx r1                                         \n"
                    "debugHardfault_address: .word debugHardfault  \n");
 }
-
+#endif
```
Comment (mkardous-silabs) [inline src/platform/silabs/Logging.cpp:434]: This is problematic

File: third_party/silabs/SiWx917_sdk.gni [hunk 2]
```diff
@@ -21,42 +21,29 @@ import("${chip_root}/src/lib/lib.gni")
...
-  enable_openthread_cli = true
 
   kvs_max_entries = 255
 
   # Use Silabs factory data provider example.
   # Users can implement their own.
   use_efr32_factory_data_provider = true
 
-  # Enable Segger System View
-  use_system_view = false
```
Comment (mkardous-silabs) [inline third_party/silabs/SiWx917_sdk.gni:42]: why is this removed? its a valuable tool to be able to use

**Review by mkardous-silabs — CHANGES_REQUESTED** [review]

File: src/platform/silabs/Logging.cpp (hunk 1 above)
Comment (jmartinez-silabs) [inline src/platform/silabs/Logging.cpp:434]: remove this

File: scripts/examples/gn_efr32_example.sh [hunk 3]
```diff
@@ -220,7 +220,15 @@ else
     BUILD_DIR=$OUTDIR/$SILABS_BOARD
     echo BUILD_DIR="$BUILD_DIR"
     if [ "$USE_WIFI" == true ]; then
-        gn gen --check --fail-on-unused-args --export-compile-commands --root="$ROOT" --dotfile="$ROOT"/build_for_wifi_gnfile.gn --args="silabs_board=\"$SILABS_BOARD\" $optArgs" "$BUILD_DIR"
+        # wifi build
```
Comment (jmartinez-silabs) [inline scripts/examples/gn_efr32_example.sh:223]: I believe this should be re-thought

When building a Siwx917 example board is always BRD4325B and it doesn't need the --wifi preset anymore.

it should use the normal build option with the right paths and right boards

File: examples/light-switch-app/silabs/SiWx917/args.gni [hunk 4]
```diff
@@ -18,6 +18,7 @@ import("${chip_root}/src/platform/silabs/SiWx917/args.gni")
```
Comment (chirag-silabs) [inline examples/light-switch-app/silabs/SiWx917/args.gni:18]: Shouldn't the same modifications are required for the other apps like lighting-app

File: examples/light-switch-app/silabs/SiWx917/build_for_wifi_args.gni [hunk 5]
```diff
@@ -19,3 +19,4 @@ chip_enable_openthread = false
 import("${chip_root}/src/platform/silabs/SiWx917/wifi_args.gni")
 
 chip_enable_ota_requestor = true
```
Comment (chirag-silabs) [inline examples/light-switch-app/silabs/SiWx917/build_for_wifi_args.gni:21]: same here, remove the build_for_wifi_args.gni and build_for_wifi_gnfile.gn and have just the .gn and args.gni

File: examples/lock-app/silabs/SiWx917/BUILD.gn [hunk 6]
```diff
@@ -150,149 +59,37 @@ efr32_sdk("sdk") {
...
   deps = [
     ":sdk",
-    "${chip_root}/examples/lock-app/lock-common",
-    "${chip_root}/examples/providers:device_info_provider",
-    "${chip_root}/src/lib",
-    "${chip_root}/src/setup_payload",
-    "${chip_root}/third_party/openthread/platforms:libopenthread-platform",
-    "${chip_root}/third_party/openthread/platforms:libopenthread-platform-utils",
-    "${examples_plat_dir}:efr-matter-shell",
+    "${examples_plat_dir}:efr32-common",
```
Comment (jmartinez-silabs) [inline examples/lock-app/silabs/SiWx917/BUILD.gn:89]: shouldn't this be `siwx917-common`

File: src/platform/silabs/SiWx917/BUILD.gn [hunk 7]
```diff
@@ -18,18 +18,27 @@ import("${chip_root}/src/platform/device.gni")
...
-
 if (chip_crypto == "platform") {
   import("//build_overrides/mbedtls.gni")
 }
+
+config("efr32-platform-wifi-config") {
+  include_dirs = [ "wifi" ]
+  defines = []
+
+  if (use_rs9116 && use_rs911x_sockets) {
```
Comment (chirag-silabs) [inline src/platform/silabs/SiWx917/BUILD.gn:34]: shouldn't this be use_siwx917

File: src/platform/silabs/SiWx917/wifi_args.gni [hunk 8]
```diff
@@ -25,7 +25,7 @@ arm_platform_config = "${efr32_sdk_build_root}/efr32_arm.gni"
 
 mbedtls_target = "${efr32_sdk_build_root}:efr32_sdk"
 
-chip_crypto = "platform"
+chip_crypto = "mbedtls"
```
Comment (chirag-silabs) [inline src/platform/silabs/SiWx917/wifi_args.gni:28]: here as well, better to remove args.gni and rename this as args.gni
Plus the chip_crypto should be platform here

File: examples/light-switch-app/silabs/SiWx917/BUILD.gn [hunk 9]
```diff
@@ -188,38 +98,21 @@ efr32_sdk("sdk") {
...
-    #sources += [ "${examples_plat_dir}/uart.cpp" ]
-  }
-
   deps = [
     ":sdk",
-    "${chip_root}/examples/light-switch-app/light-switch-common",
-    "${chip_root}/examples/providers:device_info_provider",
-    "${chip_root}/src/lib",
-    "${chip_root}/src/setup_payload",
+    "${examples_plat_dir}:efr32-common",
```
Comment (jmartinez-silabs) [inline examples/light-switch-app/silabs/SiWx917/BUILD.gn:114]: it should be `siwx917-common` i believe

**Review by jmartinez-silabs — CHANGES_REQUESTED** [review]
For SiWx917 Soc example all the 

`build_for_wifi_args.gni `and `build_for_wifi_gnfile.gni` should be remove and their content should replace the default `.gn` and `.args.gni `of the example. Those files were created to support alternating between the required structure for Thread or Wifi. 

With 917 SOC you don't have a thread option so no need to maintain both file structures.

You seem to have done the right thing for the lighting-app but not for the other apps.

File: src/platform/silabs/SiWx917/BUILD.gn (hunk 7 above)
Comment (mkardous-silabs) [reply src/platform/silabs/SiWx917/BUILD.gn:34]: add on to chirags question, why is there and if (917) when its a 917 build.gn?

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 10]
```diff
@@ -14,12 +14,49 @@
...
+
+import("${silabs_common_plat_dir}/SiWx917/args.gni")
+
+# Sanity check
+assert((chip_enable_wifi && use_SiWx917))
+if (chip_enable_wifi) {
+  assert(use_SiWx917)
+  import("${chip_root}/src/platform/silabs/SiWx917/wifi_args.gni")
+  import("SiWx917/rs911x.gni")
+}
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply examples/platform/silabs/SiWx917/BUILD.gn:57]: Yes You are right we don't require if condition here.
Asserts are required to check if the variables are set because we use these in build files which are common.

File: third_party/silabs/SiWx917_sdk.gni [hunk 11]
```diff
@@ -21,42 +21,29 @@ import("${chip_root}/src/lib/lib.gni")
...
+  use_silabs_thread_lib = false
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:36]: Yes it is not required , I missed this variable.

File: third_party/silabs/SiWx917_sdk.gni (hunk 2 above)
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:42]: Yes it is a valuable tool but currently we are not using it.
And it is using files provided by Gecko SDK to build.

So once we have system view provided in Wisemcu SDK we will add it.

File: src/platform/silabs/SiWx917/BUILD.gn [hunk 12]
```diff
@@ -18,18 +18,27 @@ import("${chip_root}/src/platform/device.gni")
...
-
-if (chip_enable_openthread) {
-  import("//build_overrides/openthread.gni")
-}
-
 if (chip_crypto == "platform") {
   import("//build_overrides/mbedtls.gni")
 }
+
+config("efr32-platform-wifi-config") {
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply src/platform/silabs/SiWx917/BUILD.gn:30]: done

File: src/platform/silabs/SiWx917/BUILD.gn (hunk 7 above)
Comment (sharad-patil24) [reply src/platform/silabs/SiWx917/BUILD.gn:34]: done 

Only used one condition use_rs911x_sockets

File: third_party/silabs/SiWx917_sdk.gni [hunk 13]
```diff
@@ -100,25 +87,32 @@ template("efr32_sdk") {
...
+      "CCP_SI917_BRINGUP",
+      "RS911X_WIFI",
+      "RSI_WLAN_ENABLE",
+      "RSI_WITH_OS",
+      "RSI_M4_INTERFACE",  #Enable CCP bus Interface
+      "RSI_WLAN_API_ENABLE",
+
+      "LITTLE_ENDIAN_ORDER",
+      "_CHIP_9118",
+    #Matter required at least 40960 but SiWx917 SoC requires an extra 20k to resolve nvm3_open() error, Need to be checked.
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:109]: done removed comment

File: third_party/silabs/SiWx917_sdk.gni (hunk 11 above)
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:36]: Removed Flag

File: third_party/silabs/SiWx917_sdk.gni [hunk 14]
```diff
@@ -130,16 +124,49 @@ template("efr32_sdk") {
...
+          "RS911X_WIFI",
+          "SiWx917_WIFI",
+          "RSI_WLAN_ENABLE",
+          "RSI_SPI_INTERFACE",
+          "RSI_WITH_OS",
+        ]
+      }
 
-      defines += [ "SL_SYSTEM_VIEW=1" ]
+    if (!use_rs911x_sockets) {
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:142]: Yes rs911x_sockets is on need basis 
Currently we use LWIP

File: third_party/silabs/SiWx917_sdk.gni [hunk 15]
```diff
@@ -130,16 +124,49 @@ template("efr32_sdk") {
...
 
-    if (use_system_view) {
-      _include_dirs += [
-        "${efr32_sdk_root}/util/third_party/segger/systemview/SEGGER",
-        "${efr32_sdk_root}/util/third_party/segger/systemview/Sample/FreeRTOSV10/Config/Cortex-M",
-        "${efr32_sdk_root}/util/third_party/segger/systemview/Sample/FreeRTOSV10",
-        "${efr32_sdk_root}/util/third_party/segger/systemview/Config/",
-      ]
+    if (defined(invoker.chip_enable_wifi) && invoker.chip_enable_wifi) {
+      if (use_SiWx917) {
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:128]: done

File: third_party/silabs/SiWx917_sdk.gni [hunk 16]
```diff
@@ -161,18 +188,13 @@ template("efr32_sdk") {
     if (chip_build_libshell) {  # matter shell
       defines += [
         "ENABLE_CHIP_SHELL",
-        "OPENTHREAD_CONFIG_CLI_TRANSPORT=OT_CLI_TRANSPORT_CONSOLE",
         "CHIP_DEVICE_CONFIG_THREAD_ENABLE_CLI=1",
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply third_party/silabs/SiWx917_sdk.gni:191]: done

File: src/platform/silabs/SiWx917/BUILD.gn (hunk 12 above)
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply src/platform/silabs/SiWx917/BUILD.gn:30]: done

File: src/platform/silabs/Logging.cpp (hunk 1 above)
Comment (sharad-patil24) [reply src/platform/silabs/Logging.cpp:434]: removed

**Review by mkardous-silabs — CHANGES_REQUESTED** [review]

File: third_party/silabs/siwx917_executable.gni [hunk 17]
```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2020 Project CHIP Authors
```
Comment (jmartinez-silabs) [inline third_party/silabs/siwx917_executable.gni:1]: I don't believe there is a reason for a duplicate here. This is what forms the binary and flashing package. I don't think there is anything different for the siwx917 (correct me if I am wrong).

What I mean when we discussed changing efr32_executable is to just rename it for all to `silabs_executable.gni`

File: scripts/examples/gn_efr32_example.sh (hunk 3 above)
Comment (jmartinez-silabs) [reply scripts/examples/gn_efr32_example.sh:223]: So I tested on your branch with the latest changes. And indeed we shouldn't need this if are any --wifi preset for siwx917 soc.

currently I can build with
`./scripts/examples/gn_efr32_example.sh examples/lighting-app/silabs/SiWx917/ out/lighting_app_siwx BRD4325B use_SiWx917=true`

And we should not need to pass `use_SiWx917=true` either. If you do need this parameter specific if in the building part we can set use_SiWx917=true  in `silabs_board.gni` for board `BRD4325B` since it is always true for that board :)

File: examples/lighting-app/silabs/SiWx917/.gn [hunk 18]
```diff
@@ -23,7 +23,7 @@ check_system_includes = true
 default_args = {
   target_cpu = "arm"
   target_os = "freertos"
-  chip_openthread_ftd = true
+  chip_enable_wifi = true
```
[the remark this answers is not in the document]
Comment (jmartinez-silabs) [reply examples/lighting-app/silabs/SiWx917/.gn:26]: this `chip_enable_wifi = true` can probably be moved to
`src/platform/silabs/siwx917/args.gni` now and be removed from every  siwx917 example's .gn

File: examples/lighting-app/silabs/SiWx917/BUILD.gn [hunk 19]
```diff
@@ -149,151 +54,33 @@ efr32_sdk("sdk") {
...
-      defines += [ "SL_WFX_CONFIG_SOFTAP" ]
-    }
-    if (sl_wfx_config_scan) {
-      defines += [ "SL_WFX_CONFIG_SCAN" ]
-    }
-  }
 }
 
+
 efr32_executable("lighting_app") {
```
[the remark this answers is not in the document]
Comment (jmartinez-silabs) [reply examples/lighting-app/silabs/SiWx917/BUILD.gn]: it should be renamed to `silabs_executable` now. This is what package the binary and the flashing pyhton. It is common to all silabs platform see `efr32_executable.gni ` and enable that file end the template inside

File: scripts/examples/gn_efr32_example.sh (hunk 3 above)
Comment (chirag-silabs) [reply scripts/examples/gn_efr32_example.sh:223]: @jmartinez-silabs true, there is actually no point of having the --wifi SiWx917 flag since it should be enabled by default. We kept this flag just to differ from the thread. --wifi flag confirms that it is a wifi specific hardware and user doesn't get confused with the thread and since we have the ncp solution of SiWx917 as well this tells that the BRD4325B board uses the SiWx917 as the wifi part. 
Would like your opinion on this.

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 20]
```diff
@@ -52,6 +52,7 @@ import("${silabs_common_plat_dir}/SiWx917/args.gni")
 assert(chip_enable_wifi)
 assert(use_SiWx917)
 import("${chip_root}/src/platform/silabs/SiWx917/args.gni")
+wisemcu_sdk_root = "${chip_root}/third_party/silabs/wisemcu-wifi-bt-sdk"
```
Comment (mkardous-silabs) [inline examples/platform/silabs/SiWx917/BUILD.gn:55]: why do you need this path here?

File: third_party/silabs/BUILD.gn [hunk 21]
```diff
@@ -35,17 +36,25 @@ declare_args() {
...
+  }
+} else {
+  assert(efr32_sdk_target != "", "efr32_sdk_target must be specified")
+  group("efr32_sdk") {
+    public_deps = [ efr32_sdk_target ]
+    public_configs = [ ":silabs_config" ]
+  }
 }
 
 if (silabs_board != "BRD4325B") {  # CCP board
```
Comment (mkardous-silabs) [inline third_party/silabs/BUILD.gn:60]: Seems this part could potentially go in the else instead of having two conditionnals?

**Review by mkardous-silabs — CHANGES_REQUESTED** [review]

File: third_party/silabs/SiWx917_sdk.gni [hunk 22]
```diff
@@ -130,17 +121,50 @@ template("efr32_sdk") {
...
+          "LWIP_ARP=1",
+          "LWIP_ICMP=1",
+          "LWIP_IGMP=1",
+          "LWIP_DHCP=1",
+          "LWIP_DNS=0",
+        ]
+      } else {
+        defines += [ "LWIP_IPV4=0" ]
+      }
+      if (lwip_ipv4) {
```
Comment (rosahay-silabs) [inline third_party/silabs/SiWx917_sdk.gni:160]: should be `lwip_ipv6`

**Review by rosahay-silabs — CHANGES_REQUESTED** [review]

File: scripts/examples/gn_efr32_example.sh (hunk 3 above)
Comment (chirag-silabs) [reply scripts/examples/gn_efr32_example.sh:223]: Modified on the latest commit, it is. The project is now building with 
`./scripts/examples/gn_efr32_example.sh examples/lighting-app/silabs/SiWx917/ out/SoC/lighting-app/ BRD4325B`

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 23]
```diff
@@ -91,6 +131,144 @@ source_set("efr32-factory-data-provider") {
...
+  }
+}
+
+config("silabs-wifi-config") {
+  defines = []
+  include_dirs = []
+
+  if (chip_default_wifi_ssid != "") {
+    defines += [
+      "CHIP_ONNETWORK_PAIRING=1",
```
[the remark this answers is not in the document]
Comment (chirag-silabs) [reply examples/platform/silabs/SiWx917/BUILD.gn]: Updated it on the latest commit

File: third_party/silabs/SiWx917_sdk.gni (hunk 22 above)
Comment (chirag-silabs) [reply third_party/silabs/SiWx917_sdk.gni]: done on the latest commit

File: third_party/silabs/siwx917_executable.gni (hunk 17 above)
Comment (chirag-silabs) [reply third_party/silabs/siwx917_executable.gni]: modified on the latest commit

File: examples/lighting-app/silabs/SiWx917/BUILD.gn (hunk 19 above)
[the remark this answers is not in the document]
Comment (chirag-silabs) [reply examples/lighting-app/silabs/SiWx917/BUILD.gn]: modified on the latest commit

File: third_party/silabs/BUILD.gn (hunk 21 above)
Comment (sharad-patil24) [reply third_party/silabs/BUILD.gn:59]: done in latest commit

File: examples/platform/silabs/SiWx917/BUILD.gn (hunk 20 above)
Comment (sharad-patil24) [reply examples/platform/silabs/SiWx917/BUILD.gn:50]: We are using wisemcu_sdk_root  variables for few files

source_set("siwx917-matter-shell") {
  if (chip_build_libshell) {
    defines = [ "ENABLE_CHIP_SHELL" ]

    sources = [
      "${silabs_common_plat_dir}/matter_shell.cpp",
      "${wisemcu_sdk_root}/platforms/si91x/drivers/cmsis_driver/UDMA.c",
      "${wisemcu_sdk_root}/platforms/si91x/drivers/cmsis_driver/USART.c",
    ]

File: src/platform/silabs/BLEManagerImpl.h [hunk 24]
```diff
@@ -25,6 +25,12 @@
 #pragma once
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
 
+#ifdef RSI_BLE_ENABLE
+#define MIN_CONNECTION_INT 45
+#define MAX_CONNECTION_INT 45
+#define SLAVE_LATENCY 0
+#define TIMEOUT 400
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply src/platform/silabs/BLEManagerImpl.h:32]: done on latest commit

File: examples/platform/silabs/SiWx917/BUILD.gn (hunk 10 above)
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply examples/platform/silabs/SiWx917/BUILD.gn]: Changed in Latest Commit

File: examples/lighting-app/silabs/SiWx917/.gn (hunk 18 above)
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply examples/lighting-app/silabs/SiWx917/.gn]: moved in latest commit

File: third_party/silabs/SiWx917_sdk.gni [hunk 25]
```diff
@@ -216,27 +195,25 @@ template("efr32_sdk") {
     cflags += [
       "-Wno-maybe-uninitialized",
       "-Wno-shadow",
+      "-Wno-empty-body",
```
Comment (jmartinez-silabs) [inline third_party/silabs/SiWx917_sdk.gni:198]: All those flag disabled to build your sdk will need to be addressed asap. Please create a task to track this.
The list is extensive and some of the flags are really important to prevent risky bugs

**Review by jmartinez-silabs — APPROVED** [review]

File: third_party/silabs/SiWx917_sdk.gni (hunk 25 above)
Comment (chirag-silabs) [reply third_party/silabs/SiWx917_sdk.gni:198]: Yeah, already discussing this with the sdk team. Most of them are addressed in the latest package which we will be getting this month end

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 26]
```diff
@@ -14,10 +14,40 @@
...
+import("${chip_root}/third_party/silabs/silabs_board.gni")
 import("${efr32_sdk_build_root}/SiWx917_sdk.gni")
+silabs_common_plat_dir = "${chip_root}/examples/platform/silabs"
+wifi_sdk_dir = "${chip_root}/src/platform/silabs/SiWx917/wifi"
+
+declare_args() {
+  enable_heap_monitoring = false
+
+  # OTA timeout in seconds
+  OTA_periodic_query_timeout = 86400
```
Comment (andy31415) [inline examples/platform/silabs/SiWx917/BUILD.gn:29]: consider adding the unit of measurement in names. Easier to see what `ota_period_query_timeout_sec` does than `ota_period_query_timeout`. I also believe OTA should be lowercase as all our args seem that way.

File: src/platform/silabs/BLEManagerImpl.h [hunk 27]
```diff
@@ -25,6 +25,12 @@
 #pragma once
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
 
+#ifdef RSI_BLE_ENABLE
+#define MIN_CONNECTION_INT 45 // 45 msec
```
Comment (andy31415) [inline src/platform/silabs/BLEManagerImpl.h:29]: for constants, consider adding unit of measurement: `MIN_CONNECTION_INTERVAL_MS` ... or even better C++ `static constexpr unsigned /* ? */ kMinBleConnectionTimeIntervalMs = 45;` (or maybe consider namespacing to make it clear these are for BLE.)

File: src/platform/silabs/BLEManagerImpl.h [hunk 28]
```diff
@@ -25,6 +25,12 @@
...
+#define MAX_CONNECTION_INT 45 // 45 msec
+#define SLAVE_LATENCY 0
+#define TIMEOUT 400
```
Comment (andy31415) [inline src/platform/silabs/BLEManagerImpl.h:32]: What is timeout and slave latency measured in? MS?

`TIMEOUT` is a very generic name, not obviously BLE. C++-namespace it for ble or name it `BLE_TIMEOUT_MS` or similar.

**Review by andy31415 — APPROVED** [review]

**Review by rosahay-silabs — APPROVED** [review]

File: src/platform/silabs/BLEManagerImpl.h (hunk 27 above)
Comment (sharad-patil24) [reply src/platform/silabs/BLEManagerImpl.h]: Changed in latest commit

File: src/platform/silabs/BLEManagerImpl.h (hunk 28 above)
Comment (sharad-patil24) [reply src/platform/silabs/BLEManagerImpl.h]: Changed in latest commit

**Review by mkardous-silabs — APPROVED** [review]

File: examples/window-app/silabs/SiWx917/args.gni [hunk 29]
```diff
@@ -16,9 +16,9 @@ import("//build_overrides/chip.gni")
...
-efr32_sdk_target = get_label_info(":sdk", "label_no_toolchain")
+siwx917_sdk_target = get_label_info(":sdk", "label_no_toolchain")
 
+app_data_model = "${chip_root}/examples/window-app/common:window-common"
 chip_enable_ota_requestor = true
-chip_enable_openthread = true
-openthread_external_platform =
-    "${chip_root}/third_party/openthread/platforms/efr32:libopenthread-efr32"
+
+use_base_app = false
```
[the remark this answers is not in the document]
Comment (jmartinez-silabs) [reply examples/window-app/silabs/SiWx917/args.gni:24]: The window cover app is structured differently than other apps. It is the only app that doesn't use base_applications.cpp/h. This args is used to include or not that source file

File: third_party/silabs/SiWx917_sdk.gni [hunk 30]
```diff
@@ -87,41 +64,40 @@ template("efr32_sdk") {
...
       "KVS_MAX_ENTRIES=${kvs_max_entries}",
-      "EFR32_OPENTHREAD_API",
-      "PHY=EMBER_PHY_RAIL",
-      "CORTEXM3",
-      "MICRO=EMBER_MICRO_CORTEXM3_EFR32",
-      "PLAT=EMBER_PLATFORM_CORTEXM3",
       "${silabs_mcu}=1",
       "${silabs_board}=1",
       "__HEAP_SIZE=0",
-      "SL_COMPONENT_CATALOG_PRESENT",
```
[the remark this answers is not in the document]
Comment (jmartinez-silabs) [reply third_party/silabs/SiWx917_sdk.gni:124]: 917 isn't gsdk so no, not at the moment

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 31]
```diff
@@ -69,12 +99,19 @@ if (enable_dic) {
...
+source_set("siwx917-matter-shell") {
   if (chip_build_libshell) {
     defines = [ "ENABLE_CHIP_SHELL" ]
 
-    sources = [ "${silabs_common_plat_dir}/matter_shell.cpp" ]
-    include_dirs = [ "." ]
+    sources = [
+      "${silabs_common_plat_dir}/matter_shell.cpp",
+      "${wisemcu_sdk_root}/platforms/si91x/drivers/cmsis_driver/UDMA.c",
+      "${wisemcu_sdk_root}/platforms/si91x/drivers/cmsis_driver/USART.c",
```
[the remark this answers is not in the document]
Comment (chirag-silabs) [reply examples/platform/silabs/SiWx917/BUILD.gn:109]: Yeah
